### PR TITLE
SITES-593 tool to make a static site snapshot

### PR DESCRIPTION
**Pull request by freshtonic**

### CODE REVIEW

File: config/environments/static.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  config.public_file_server.enabled = true
+  config.assets.js_compressor = :uglifier
+  config.assets.debug = false
+
+  Rails.application.routes.default_url_options = {
+      protocol: 'http',
+      host: 'localhost:3001'
+  }
+
+  config.content_analysis_base_url = "https://dummy"
```
Comment (bjgordon) [inline config/environments/static.rb]: This line is also in `external_apis.rb`, can we delete it from here?

File: lib/spider.rb [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+      unless depth >= @max_depth
+        non_asset_links(nodes).each do |link|
+          crawl(URI.join(uri, link), depth + 1) # rescue nil
+        end
+      end
+    end
+
+    def external?(uri)
+      uri.host && uri.host != @root_uri.host
```
Comment (bjgordon) [inline lib/spider.rb]: /pedantic should be a case insensitive comparison

**Comment by bjgordon** [comment]
Minor comments aside, LGTM.

File: config/environments/static.rb (hunk 1 above)
Comment (freshtonic) [inline config/environments/static.rb]: yes this can be deleted

**Comment by ltankey** [comment]
Should we let environment variables set the target ports? It may be simpler for us to provide the option in order to play nice with a CF environment, unless any potential CF deploy would rely on nginx acting as a reverse proxy?

**Comment by maxious** [comment]
I saw that more as a way to test that the crawl worked correctly rather than any way to host it publically

**Comment by freshtonic** [comment]
@ltankey I envisage this being executed on a developer laptop with a snapshot of production. Not sure we need that - and if we do we can add it later.

**Comment by ltankey** [comment]
@freshtonic ah I see, makes sense then. LGTM

**Pull request merged**